libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1512,7 +1525,7 @@ std::string YulUtilFunctions::decreaseByteArraySizeFunction(ArrayType const& _ty
 					case 1 {
 						let arrayDataStart := <dataPosition>(array)
 						// clear whole old array, as we are transforming to short bytes array
```

Comment:
This comment is not incorrect, but for a moment it made me think we have a bug because `clearStorageRange()` omits the first slot. I'd clarify it:

```suggestion
						// Clear whole old array, as we are transforming to short bytes array.
						// <transitLongToShort>() will clear the first slot after copying its content.
```